src/RequestHandler/RequestManager.js

```diff
@@ -0,0 +1,40 @@
+const {HandlerFactory} = require('./HandlerFactory');
+const supportedProtocols = ['mqtt'];
+function RequestManager () {
+  const handlersList = {};
+
+  async function createReqHandler(dataFromParser) {
+    for (const [serverName,serverData] of Object.entries(dataFromParser.servers)) {
+      const handlerInstance = await HandlerFactory(serverData , dataFromParser.PublishOperations).catch((err) => {
+        console.log(err.message);
+        return 'notHealthy';
```

Comment:
You will have to revise this implementation, you cannot return inside a `catch` statement like so. You return `notHealthy` to the catch callback, which does not resolve to `handlerInstance`.

Reply:
i tested it and it did. i debugged that line  inserted in serverData.protocol a random 'lel' string. and then handlerinstance was 'nothealthy'

Reply:
Huh, yea you are right 🧐 Did not know that was possible actually.

Regardless, I don't think we should mix await syntax with promise handling, stay with one specific. Also, this returning of strings is a bit weird.

The rewrite of this could be something like:

```js
try {
         handlersList[parseInt(serverName, 10)] = await HandlerFactory(serverData , dataFromParser.PublishOperations);
}catch(e) {return}
```

It makes it more clean in my opinion 🙂

Reply:
Or even better, depends on preference, you could still return undefined from `HandlerFactory` and then add an if statement checking the return value instead of throwing an exception. 🙂